enclave/core/sgx/init.c

```diff
@@ -69,12 +69,14 @@ static oe_result_t _eeid_patch_memory()
         uint8_t* enclave_base = (uint8_t*)__oe_get_enclave_base();
         uint8_t* heap_base = (uint8_t*)__oe_get_heap_base();
         oe_eeid_marker_t* marker = (oe_eeid_marker_t*)heap_base;
-        uint8_t* eeid_ptr = enclave_base + marker->offset;
+        oe_eeid_t* eeid = (oe_eeid_t*)(enclave_base + marker->offset);
 
-        if (!_is_within_eeid_enclave(eeid_ptr))
+        if (!_is_within_eeid_enclave((uint8_t*)eeid) ||
+            !_is_within_eeid_enclave(eeid->data) ||
+            !_is_within_eeid_enclave(eeid->data + eeid->data_size))
```

Comment:
I think eeid->data_size does not count the SIGSTRUCT at the end of the data[] buffer.

I just realized that the code should also check to make sure the eeid page location the marker->offset claims does not overlap with the heap or the TCS/Stack pages.

Reply:
Would an overlap with the TCS/stack pages be a security problem? It's easy to add a check that it's within heap + eeid size, so I can add that anyways.

Reply:
(Yes, done)